File: Data Structures & Algorithms/counting-bits/submission-6.cpp
class Solution {
   public:
    vector<int> countBits(int n) {
        vector<int> result(n + 1, 0);

        // Iterate through 1 to n
        for (int num = 1; num <= n; ++num) {
            // Iterate through all 32 bits in num
            for (int i = 0; i < 32; i++) {
                // Increment result[num] if ith bit in num is 1
                if (num & (1 << i)) {
                    result[num]++;
                }
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/majority-element/submission-6.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        result = 0
        count = 0

        for num in nums:
            if count == 0:
                result = num
            
            if result != num:
                count -= 1
            else:
                count += 1
        return result

